crates/sui-move-build/src/linters/mod.rs

```diff
@@ -128,3 +134,21 @@ pub fn base_type(t: &N::Type) -> Option<&N::Type> {
         T::Unit | T::Var(_) | T::Anything | T::UnresolvedError => None,
     }
 }
+
+/// Returns true if a given type represents an object (or a wrappable object if the argument is set
+/// to true).
+pub fn is_obj_type(sp!(_, st_): &SingleType, wrappable: bool) -> bool {
```

Comment:
might be better to do something like 
```suggestion
pub fn is_obj_type(sp!(_, st_): &SingleType) -> Option<AbilitySet> {
```
Where the set is Some if the type is an object, and None otherwise. Otherwise we might just start threading through more and more flags